xcdat/spatial.py

```diff
@@ -244,42 +245,36 @@ def get_weights(
         and pressure).
         """
         Bounds = TypedDict(
-            "Bounds",
-            {"domain": xr.DataArray, "region": Optional[RegionAxisBounds]},
+            "Bounds", {"weights_method": Callable, "region": Optional[np.ndarray]}
         )
-        axis_bounds: Dict[SpatialAxis, Bounds] = {}
-        if "X" in axis:
-            axis_bounds["X"] = {
-                "domain": self._dataset.bounds.get_bounds("X").copy(),
-                "region": lon_bounds,
-            }
-        if "Y" in axis:
-            axis_bounds["Y"] = {
-                "domain": self._dataset.bounds.get_bounds("Y").copy(),
-                "region": lat_bounds,
-            }
 
-        axis_weights: AxisWeights = {}
+        axis_bounds: Dict[SpatialAxis, Bounds] = {
+            "X": {
+                "weights_method": self._get_longitude_weights,
+                "region": np.array(lon_bounds, dtype="float")
+                if lon_bounds is not None
+                else None,
+            },
+            "Y": {
+                "weights_method": self._get_latitude_weights,
+                "region": np.array(lat_bounds, dtype="float")
+                if lat_bounds is not None
+                else None,
+            },
+        }
 
+        axis_weights: AxisWeights = {}
         for key in axis:
-            d_bounds = axis_bounds[key]["domain"]
-            # the logic for generating longitude weights depends on the
-            # bounds being ordered such that d_bounds[:, 0] < d_bounds[:, 1]
-            # they are re-ordered (if need be) for the purpose of creating
-            # weights
+            d_bounds = self._dataset.bounds.get_bounds(key).copy()
+            # The logic for generating longitude weights depends on the
+            # bounds being ordered such that d_bounds[:, 0] < d_bounds[:, 1].
+            # They are re-ordered (if need be) for the purpose of creating
+            # weights.
             d_bounds = self._force_domain_order_low_to_high(d_bounds)
 
-            r_bounds: Union[Optional[RegionAxisBounds], np.ndarray] = axis_bounds[key][
-                "region"
-            ]
-            if r_bounds is not None:
-                r_bounds = np.array(r_bounds, dtype="float")
-
-            if key == "X":
-                weights = self._get_longitude_weights(d_bounds, r_bounds)
-            elif key == "Y":
-                weights = self._get_latitude_weights(d_bounds, r_bounds)
+            r_bounds = axis_bounds[key]["region"]
 
+            weights = axis_bounds[key]["weights_method"](d_bounds, r_bounds)
             weights.attrs = d_bounds.attrs
             axis_weights[key] = weights
```

Comment:
I refactored a bit here to reduce `if else` statements and avoids fetching `domain_bounds` for unspecified keys in `axis`.

Reply:
This commented was intended to highlight the entire method, not just this line.